src/main/java/seedu/address/model/person/JobApplied.java

```diff
@@ -0,0 +1,56 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a candidate's job applied in HR+.
+ * Guarantees: immutable; is valid as declared in {@link #isValidJobApplied(String)}
+ */
+public class JobApplied {
+    public static final String MESSAGE_JOB_APPLIED_CONSTRAINTS =
+            "Job applied can take any values, and it should not be blank";
+
+    /**
+     * The first character of the job applied must not be a whitespace,
+     * otherwise " " (a blank string) becomes a valid input.
+     */
+    public static final String JOB_APPLIED_VALIDATION_REGEX = "[^\\s].*";
```

Comment:
Regex should consistent with class `Name` and `Major`. This regex is similar to class `Address` which allows special characters too.

Reply:
Thanks for reviewing! My rationale is that using the regex similar to `Name` and `Major` will disallow users to put in titles like "Front-end Developer" or "C++ Developer"? What do the rest think?

Reply:
That's very true. We can keep it that way then :smile: